### DISCUSSION TITLE
When is `NitroStaticBuildFlags.dev` undefined?

### DISCUSSION THREAD

**Discussion by septatrix** [nitrojs/nitro#2901] [Q&A]
The types state that the option can be undefined which would make testing for `if (!import.meta.dev)` become true when `dev` is unset. This could have unintended consequences. Under which circumstances is the `dev` property missing? Wouldn't it make more sense to have it default to `false` in such cases

**Comment by pi0** [comment]
In js `!false === !undefined` ~> Essentially both possible values for `import.meta.dev` have same effect in your `if (!import.meta.dev)` example.

Can you please elaborate more if I'm missing something?

**Reply by septatrix** [reply]
You are right, the comparison still works. I got worried because some (custom?) eslint rule in our project flags it if `boolean | null | undefined` is negated. I was curious under which circumstances it could ever be `undefined` and why it is typed that way instead of simply having it be a bool.

**Reply by pi0** [reply]
It is because nitro rollup build only **sets** it in certain cases, in other envs, it won't statically replace `import.meta.dev` at all for perf. so it is either not set at all during build or `true`.

**Reply by septatrix** [reply]
Shouldn't it by typed as `true | undefined` then?

**Reply by pi0** [reply]
You are right. Can you please make a PR to fix types? 🙏🏼

**Reply by septatrix** [reply]
Changing the type does not seem to work because one may write `dev: false` in their Nitro options
